internal/util/projutil/interactive_promt_util.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package projutil
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"strings"
+)
+
+// InteractiveLevel captures the user preference on the generation of interactive
+// commands.
+type InteractiveLevel int
+
+const (
+	// User has not turned interactive mode on or off, default to off.
+	InteractiveSoftOff InteractiveLevel = iota
+	// User has explicitly turned interactive mode off.
+	InteractiveHardOff
+	// User only explicitly turned interactive mode on.
+	InteractiveOnAll
+)
+
+func GetRequiredInput(msg string) string {
+	return getRequiredInput(os.Stdin, msg)
+}
+
+func getRequiredInput(rd io.Reader, msg string) string {
+	reader := bufio.NewReader(rd)
+
+	for {
+		fmt.Printf("%s", msg)
+		value := readInput(reader)
+		if value != "" {
+			return value
+		}
+		fmt.Printf("Input is required. ")
+	}
+}
+
+func GetOptionalInput(msg string) string {
+	fmt.Printf("%s", msg)
+	value := readInput(bufio.NewReader(os.Stdin))
+	return value
+}
+
+func GetStringArray(msg string) []string {
+	return getStringArray(os.Stdin, msg)
+}
+
+func getStringArray(rd io.Reader, msg string) []string {
+	reader := bufio.NewReader(rd)
+	for {
+		fmt.Printf("%s", msg)
+		value := readArray(reader)
+		if len(value) != 0 && len(value[0]) != 0 {
+			return value
+		}
+		fmt.Printf("No list provided. ")
+	}
+}
+
+// readstdin reads a line from stdin and returns the value.
+func readstdin(reader *bufio.Reader) string {
```

Comment:
This isn't stdin-specific is it?
```suggestion
func readLine(reader *bufio.Reader) string {
```